### PR TITLE
NDRCN-284 Changes

### PR DESCRIPTION
**Pull request by somaraja**

### CODE REVIEW

File: app/forms/ImporterEoriFormProvider.scala [hunk 1]
```diff
@@ -26,6 +26,11 @@ class ImporterEoriFormProvider @Inject() extends Mappings {
   def apply(): Form[EORI] =
     Form(
       "value" -> text("importerEori.error.required")
-        .verifying(maxLength(17, "importerEori.error.length")).transform[EORI](EORI.apply, _.value)
+        .verifying(minLength(14, "importerEori.error.length"))
+        .verifying(maxLength(14, "importerEori.error.length"))
```
Comment (lmugnai) [inline app/forms/ImporterEoriFormProvider.scala:30]: Hi @somaraja Should we not set the max to 17? I think the user could add an EORI with GB follwed by either 12 or 15 digits. Can we discuss this tomorrow?

File: app/forms/Validation.scala [hunk 2]
```diff
@@ -22,6 +22,7 @@ object Validation {
 
   val accountNumberPattern: Regex   = "^[ -]*(?:\\d[ -]*){6,8}$".r.anchored
   val sortCodePattern: Regex        = "^[ -]*(?:\\d[ -]*){6,6}$".r.anchored
+  val eoriPattern: Regex = "(?:[Gg][Bb])?\\d{12}".r.anchored
```
Comment (lmugnai) [inline app/forms/Validation.scala:25]: The regex example we have inthe api doc is 
"([a-zA-Z]{2}[0-9]{15})|([a-zA-Z]{2}[0-9]{12})|([a-zA-Z]{4})" I think we should restrict the characters to 'GB/gb' as you've done but the length is different. Can we discuss?

File: test/forms/ImporterEoriFormProviderSpec.scala [hunk 3]
```diff
@@ -23,7 +23,7 @@ class ImporterEoriFormProviderSpec extends StringFieldBehaviours {
 
   val requiredKey = "importerEori.error.required"
   val lengthKey = "importerEori.error.length"
-  val maxLength = 17
+  val maxLength = 14
```
Comment (lmugnai) [inline test/forms/ImporterEoriFormProviderSpec.scala:26]: think this may need to be 17

File: test/forms/ImporterEoriFormProviderSpec.scala [hunk 4]
```diff
@@ -37,13 +37,6 @@ class ImporterEoriFormProviderSpec extends StringFieldBehaviours {
       stringsWithMaxLength(maxLength)
     )
 
-    behave like fieldWithMaxLength(
```
Comment (lmugnai) [inline test/forms/ImporterEoriFormProviderSpec.scala:40]: Why have we removed this?

**Review by lmugnai — COMMENTED** [review]
Are there no url changes we need to do as part of this ticket?

File: app/forms/Validation.scala (hunk 2 above)
Comment (somaraja) [reply app/forms/Validation.scala:25]: @lmugnai as advised I have done other changes, but not sure how to work with regex from API, will talk tomorrow.

File: test/forms/ImporterEoriFormProviderSpec.scala (hunk 3 above)
Comment (somaraja) [reply test/forms/ImporterEoriFormProviderSpec.scala]: @lmugnai incorporated the review comments

File: app/forms/ImporterEoriFormProvider.scala [hunk 5]
```diff
@@ -26,6 +26,10 @@ class ImporterEoriFormProvider @Inject() extends Mappings {
   def apply(): Form[EORI] =
     Form(
       "value" -> text("importerEori.error.required")
-        .verifying(maxLength(17, "importerEori.error.length")).transform[EORI](EORI.apply, _.value)
+        .verifying(firstError(
+          maxLength(17, "importerEori.error.length"),
```
Comment (lmugnai) [inline app/forms/ImporterEoriFormProvider.scala:30]: @somaraja if the error message is the same for but of these validation checks, then I don't think we need the maxLength check as the regexp check will check the length too.

**Review by lmugnai — APPROVED** [review]

**Pull request merged**